compiler/rustc_mir_transform/src/check_unsafety.rs

```diff
@@ -555,15 +571,21 @@ pub fn check_unsafety(tcx: TyCtxt<'_>, def_id: LocalDefId) {
                 tcx.sess.emit_err(errors::RequiresUnsafe {
                     span: source_info.span,
                     enclosing,
-                    details,
+                    details: unsafe_details,
                     op_in_unsafe_fn_allowed,
+                    help: details.help_missing_features(),
+                    note_missing_features: details.note_missing_features(),
```

Comment:
You don't need to store this any more, you can just call `note_missing_features()` where you'd be using the `note_missing_features` field. Same goes for the `help` field.